src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -89,10 +94,29 @@ protected async Task<IActionResult> ListDetectors(TResource resource, string que
             {
                 return BadRequest("Search query term should be at least two characters");
             }
-            return Ok(await this.ListDetectorsInternal(cxt, queryText));
+
+            IEnumerable<DiagnosticApiResponse> listDetectorsResponse = await this.ListDetectorsInternal(cxt, queryText);
+            IEnumerable<DiagnosticApiResponse> listDetectorsTranslatedResponse = listDetectorsResponse;
+
+            try
+            {
+                if (IsLocalizationApplicable(language))
+                {
+                    listDetectorsTranslatedResponse = await _diagnosticTranslator.GetMetadataTranslations(listDetectorsResponse, language).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                // Log translation exceptions and return original untranslated response
+                DiagnosticsETWProvider.Instance.LogRuntimeHostHandledException(cxt.OperationContext.RequestId, "ListDetectorsTranslations", cxt.OperationContext.Resource.SubscriptionId,
```

Comment:
Is there something we're going to mention in the UI if the translation is not successful? Like still render results but have a small note somewhere saying `Sorry failed to translate response, please try again later.`

Reply:
I think in case of exception, Cindy is falling back to old behavior (non-localized response) which is the right behavior.